Neos.Flow/Classes/Http/CacheControlDirectives.php

```diff
@@ -0,0 +1,195 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Http;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+/**
+ * Cache-Control HTTP header generation/parsing
+ */
+class CacheControlDirectives
+{
+    /**
+     * @var array
+     */
+    protected $cacheDirectives = [
+        'visibility' => '',
+        'max-age' => '',
+        's-maxage' => '',
+        'must-revalidate' => '',
+        'proxy-revalidate' => '',
+        'no-store' => '',
+        'no-transform' => ''
+    ];
+
+    /**
+     * @param string $rawHeaderValue The value of a specification compliant Cache-Control header
+     */
+    private function __construct(string $rawHeaderValue)
+    {
+        foreach (array_keys($this->cacheDirectives) as $key) {
+            $this->cacheDirectives[$key] = '';
+        }
+        preg_match_all('/([a-zA-Z][a-zA-Z_-]*)\s*(?:=\s*(?:"([^"]*)"|([^,;\s"]*)))?/', $rawHeaderValue, $matches, PREG_SET_ORDER);
+        foreach ($matches as $match) {
+            if (isset($match[2]) && $match[2] !== '') {
+                $value = $match[2];
+            } elseif (isset($match[3]) && $match[3] !== '') {
+                $value = $match[3];
+            } else {
+                $value = null;
+            }
+            $this->setDirective(strtolower($match[1]), $value);
+        }
+    }
+
+    /**
+     * Internally sets the cache directives correctly by parsing the given
+     * Cache-Control header value.
+     *
+     * @param string $rawHeaderValue The value of a specification compliant Cache-Control header
+     * @return self
+     */
+    public static function fromRawHeader(string $rawHeaderValue): self
+    {
+        return new self($rawHeaderValue);
+    }
+
+    /**
+     * Sets a special directive for use in the Cache-Control header, according to
+     * RFC 2616 / 14.9
+     *
+     * @param string $name Name of the directive, for example "max-age"
+     * @param string|null $value An optional value
+     * @return void
+     * @api
+     */
+    public function setDirective(string $name, ?string $value = null): void
+    {
+        switch ($name) {
+            case 'public':
+                $this->cacheDirectives['visibility'] = 'public';
+            break;
+            case 'private':
+            case 'no-cache':
+                $this->cacheDirectives['visibility'] = $name . (!empty($value) ? '="' . $value . '"' : '');
```

Comment:
TIL that *response* `no-cache` and `private` directives may specify a list of "fields" (headers, https://datatracker.ietf.org/doc/html/rfc2616#page-110) that may not be returned in the cached response without revalidation as opposed to the case without fields, where the whole response may not be returned without revalidation.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#no-cache is not comprehensive on this part